### ISSUE TITLE
Add HISAT2

### ISSUE THREAD

**Issue by ihodes** [hammerlab/biokepi#140]
Lower priority than other items on this milestone.

(:octocat: issues can't capture this nicely)

**Comment by arahuja** [comment]
@ihodes @smondet 
I started working on this tonight, but I'm unsure how to proceed for a tool upgrade or new version

1. Should this be a separate 'version' of Hisat as was done for prototype ala BWA: ` | `Hisat of [`V_0_1_6_beta | `V_2_0_2_beta]`
1. On install/setup right now there is only the default toolkit, how would you go about selecting a different version - adding an argument to `hisat_tool ~host ~meta_playground` and then just selecting based on it?
```
    | `V_0_1_6_beta -> "http://ccb.jhu.edu/software/hisat/downloads/hisat-0.1.6-beta-Linux_x86_64.zip"
    | `V_2_0_2_beta -> "ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.0.2-beta-Linux_x86_64.zip"
```
1. Once you have them setup, the binaries for each are different `hisat` vs `hisat2` The input arguments are actually the same so should we keep this separation or just alias the installed version (should a user be able to have 2 installed?) `ensure` for example doesn't currently take a version?
1. This could obviously be done as a completely separate tool - is that better or set a bad precedent for anything else that changes version? Samtools was recently upgraded in biokepi without marking a version change
1.Should we store the version in the final or only for major releases? If so, should we go back and add v1 to output hisat-v1?
1. In `pipeline.ml` there doesn't seem to be a way to select version as well, since all tools only have one currently?

**Comment by ihodes** [comment]
We chatted offline about this: tl;dr we need to version (via variant) a field in the `Configuration.t` of a given tool; we can dispatch off of this variant field in the `run` function of a given tool.

**Comment by arahuja** [comment]
Closed by #219
